GTC-2734 Add JRC/SBTN, confidence levels, and total_ha to integrated alerts

danscales opened the pull request:
GTC-2734 Add JRC/SBTN, confidence levels, and total_ha to integrated alerts

For daily alerts, add new columns that divide the daily alerts into a 3x3 cross-product of (whole area, SBTN forest only, JRC forest only) x (lo, med, hi) alert confidence

Also, add a new 'total_ha' column which is the area of the location. This area will be used by the front-end to calculate measures such as "daily alerts per hectare".

Currently, I am calculating the entire location area in the usual per-pixel way with Geodesy.pixelArea. We were already calculating the area per pixel whenever isTreeCoverExtent30 was true (more than 30% tree cover). From a look at some input sets, it seems like this is usually a large part of the location areas. So, doing the area calculation for the entire geometry shouldn't be much more expensive.

However, I will make sure that the isTreeCoverExtent30 area is a large part of the whole area for most datasets, once this change is in. If it is not, then it may give better performance to calculate the total area of locations outside of the pixel function using an equal-area projection (for which I already have some working code).

danscales committed:
GTC-2734 Add JRC/SBTN, confidence levels, and total_ha to integrated alerts.

For daily alerts, add new columns that divide the daily alerts into a
3x3 cross-product of (whole area, SBTN forest only, JRC forest only) x
(lo, med, hi) alert confidence

Also, add a new 'total_ha' column which is the area of the location.
This area will be used by the front-end to calculate measures such as
"daily alerts per hectare".

Currently, I am calculating the entire location area in the usual
per-pixel way with Geodesy.pixelArea. We were already calculating the
area per pixel whenever isTreeCoverExtent30 was true (more than 30% tree
cover). From a look at some input sets, it seems like this is usually a
large part of the location areas. So, doing the area calculation for the
entire geometry shouldn't be much more expensive.

However, I will make sure that the isTreeCoverExtent30 area is a large
part of the whole area for most datasets, once this change is in. If it
is not, then it may give better performance to calculate the total area
of locations outside of the pixel function using an equal-area
projection (for which I already have some working code).

jterry64 commented on src/main/scala/org/globalforestwatch/summarystats/gfwpro_dashboard/GfwProDashboardData.scala:
```diff
@@ -15,10 +15,25 @@ case class GfwProDashboardData(
+  /** Integrated alert count within location geometry grouped by day, and further
+    * subdivided by cross-product of cover area (all, SBTN area only, JRC area only)
+    * x alert confidence (lo, med, hi) */
+  glad_alerts_daily_all_lo: GfwProDashboardDataDateCount,
```
Since you're adding tons of new fields with "glad" in the name, but we technically switched to integrated alerts, should we use this opportunity to switch the name to use "integrated_alerts" or generically "deforestation_alerts"? At least for the new fields, this way Pro won't need to migrate to a new schema later

danscales replied:
OK, good idea.  I'll check with Kinshuk if it is ok to change the names for the new columns (I'm assuming it will be fine).  Also, I'll see if we can actually switch the names of the existing columns this time around - but the requirement is that we synchronize the updates to geotrellis and the GFWPro App code, so that is trickier to set up.

Thanks for the review!

danscales replied:
OK, I did the rename of the new columns.  I also duplicated the glad_alerts_coverage column to integrated_alerts_coverage, as suggested by Kinshuk.  The plan is that GFWPro will eventually only need the daily alerts (as broken down into all the sub-categories in this change).  So, GFWPro should eventually be able to transition to depending only on the integrated_alerts_* columns.  Once it has fully transitioned, we should be able to asynchronously remove all the glad_alerts_* columns.  Thanks for the comment!

danscales committed:
Rename new glad_alerts_* columnms to integrated_alerts_*.

Changes based on Justin's comments.

Also, I duplicated the glad_alerts_coverage column to
integrated_alerts_coverage, based on Kinshuk's suggestion. He is hoping
that we can just remove all the glad_alerts_* columns asynchronously
once the UI has switched to using only the integrated_alerts_* columns.
(The weekly and month alerts won't be needed anymore).

danscales committed:
Rename alert levels to (nominal, high, highest)

This is in accordance with terminology for Flagship (though "nominal" is
sometimes specified as "Detected by a single alert system").

Changed both the enum and the associated column names.
